### ISSUE TITLE
Problem with saving in Arethusa

### ISSUE THREAD

**Issue by gcelano** [latin-language-toolkit/arethusa#608]
I am using the staging version with smyth3 tagset

When I deselect a morphological analysis and select a new one, it may happen that the new selection is not saved, but the old one is simply deleted (I saw this in the xml file). Unfortunately, this error seems to be unpredictable, in the sense that it does not apply to all changes.

**Comment by balmas** [comment]
I haven't been able to reproduce this but some more details from an email discussion with @gcelano 

"This is another typical scenario:
I have a morphological analysis which is selected (because I first annotated that way). I select a different option among the ones proposed and I go ahead with another word...but when I get back to check if everything is fine, I do not find the option ticked and not even the lemma translation (i.e., blanck field), even though it seems that they are saved in the xml."

(I think the lemma translation problem was separate and is indeed fixed now)

**Comment by gcelano** [comment]
A quick update:

The scenario is similar: Marcel annotated something. I corrected his morphological analysis (which were previously selected and saved by him): I simply selected a new morphological analysis and saved. What happened when I refreshed the browser was that no option was selected. I saw the XML and I found that the content of the attribute pos was empty. So I assume that when I selected and saved the new option, the system simply deleted the old selection but did not substitute it with the new one.

**Comment by LFDM** [comment]
Hey @gcelano, I also have not been able to reproduce this. What do you mean with "Marcel annotated" something? Are you working on a file that's not your own?

**Comment by gcelano** [comment]
I meant that he was annotating a normal file that we put in Arethusa. To
reproduce this you should select a morphological option save, refresh and
then try with changing the previous selection. I can tell that for some
strange reason this is sometimes possible but some other times it happens
that by modifying the previous morphological annotation the system simply
deletes the content of the pos attribute. This means that when you come
back/refresh the web page you find nothing selected. At that point you need
to select something again and this usually solves the problem.

2015-02-01 14:48 GMT+01:00 Gernot Höflechner <notifications@github.com>:

> Hey @gcelano <https://github.com/gcelano>, I also have not been able to
> reproduce this. What do you mean with "Marcel annotated" something? Are you
> working on a file that's not your own?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/latin-language-toolkit/arethusa/issues/608#issuecomment-72365319>
> .
>



-- 
Dr. Giuseppe G. A. Celano
Postdoctoral Researcher
Universität Leipzig
Institute of Computer Science,
​Section of ​
Digital
​ ​
Humanities
​

Augustusplatz 10
04109 Leipzig
​ (DE)​

E-mail: giuseppegacelano@gmail.com
            giuseppe.celano@tufts.edu
            celano@informatik.uni-leipzig.de
Web site 1: http://www.dh.uni-leipzig.de/wo/team/
Web site 2: https://sites.google.com/site/giuseppegacelano/

**Comment by LFDM** [comment]
Is this happening with specific words only? Ideally you would have a very concrete example for me, because when I do the steps you just described, everything works fine for me :(

**Comment by gcelano** [comment]
Unfortunately, when I find an example I cannot reproduce it anymore... and
as I was saying I have not been able to understand why this happens with
some words only, without a clear explanation :( But I will keep you
informed if I can say more about this.

2015-02-01 15:09 GMT+01:00 Gernot Höflechner <notifications@github.com>:

> Is this happening with specific words only? Ideally you would have a very
> concrete example for me, because when I do the steps you just described,
> everything works fine for me :(
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/latin-language-toolkit/arethusa/issues/608#issuecomment-72366112>
> .
>



-- 
Dr. Giuseppe G. A. Celano
Postdoctoral Researcher
Universität Leipzig
Institute of Computer Science,
​Section of ​
Digital
​ ​
Humanities
​

Augustusplatz 10
04109 Leipzig
​ (DE)​

E-mail: giuseppegacelano@gmail.com
            giuseppe.celano@tufts.edu
            celano@informatik.uni-leipzig.de
Web site 1: http://www.dh.uni-leipzig.de/wo/team/
Web site 2: https://sites.google.com/site/giuseppegacelano/

**Comment by balmas** [comment]
​This is the link I am working on : http://www.perseids.org/tools/arethusa/app/#/perseids?chunk=28&doc=9258

In the sentence there the 6th word (pole/mous) had a wrong morphological annotation. I selected the right annotation and I have not saved yet. If I go on with the annotation and then get back (without saving ever), the annotation of the word pole/mous is never in place (and in the xml I find an empty pos). I infer that when I first selected the right option the value of the attribute pos was deleted but was not replaced with the correct one. I know that if I save, refresh, and then select the right option, in the end it gets saved... but I need to do this for too many words...since I am correcting an old annotation. Is this of any help?

**Comment by balmas** [comment]
Continued problems reported by @gcelano from one of his students:

" I've been playing around with the Lemma Translations a little bit, and the problem does seem to be when you change the morphological tab.  The definitions weren't affected by me changing the syntactic or semantic tab.  Also, when you change the morphological tab the definition doesn't disappear right away, but only after you load a different page and then come back to it.  If, however, you simply save and refresh the page after changing the morphological tab, the definition will remain."

**Comment by balmas** [comment]
@Masoumeh this would be a good item to go through with @gcelano -- please ask him to show you the behavior

**Comment by Masoumeh** [comment]
As I checked, the problem appears when the changes are made in both "Lemma Translation" and checkbox together such that first the "lemma translation" is changed and then the checkbox!!!
It is reflected just in saving the "Lemma Translation" in both GUI and XML with keeping the latest value and doesn't effect the checkbox.
This doesn't happen when one of the lemma or checkboxes is changed. 
Also, refreshing the page doesn't seem to make a problem. The behavior is the same.

**Comment by balmas** [comment]
@Masoumeh do you think you can tackle fixing this?

**Comment by Masoumeh** [comment]
@balmas I think there is a relation between the forms and the lemma translation. I will work on it and inform you how it goes further.

**Comment by Masoumeh** [comment]
Hi @balmas,
How can I load the "smyth3" format locally?
I mean, how to open the setting with lemma translation in morph tab locally? Thanks.

**Comment by balmas** [comment]
Here are the steps to adding other formats. I think these should be added to the arethusa documentation but I'm not sure where they go really.

The path to auxiliary configuration files is set via the following arethusa configuration setting

main.auxConfPath 

You can find the auxiliary configuration files themselves in the [arethusa-configs](https://github.com/latin-language-toolkit/arethusa-configs) repository. 

You can clone and build them, and then copy them to wherever your auxConfPath points, or you could just point that at either the location they are currently deployed on Perseids 

http://services.perseids.org/arethusa-configs

Or in GitHub, although I think it's not guaranteed right now that the github dist folder is completely up to date, as the process of building is not completely automated right now.

See related item #661

**Comment by Masoumeh** [comment]
Actually, I don't know how to access the setting via URL.
After adding the config files you mentioned, the URL with staging config (i.e. http://localhost:8081/app/#/staging?doc=1) shows the old setting, even when I change the "format" attribute to "smyth3" in example data files (doc). 

It happens that I can see other changes (like SG tab) when using some example files with format attribute changed to 'smyth3' (like smyth2.xml), but lemma translation doesn't appear.
I also tried to put the "perseids" config in the URL. But I get the access denied error. 

Could you please explain how can I see "lemma translation" in a page locally? And why I get access denied error when using perseids config (should I add ssh key)? Thanks.

**Comment by balmas** [comment]
Hi @Masoumeh the triggers to enable the lemmaTranslation field for the morph plugin are:

1. the use of the morph3.html template
2. the `gloss` property set to true 

E.g.

```
"morph" : {
      "retrievers" : {
        "BspMorphRetriever" : {
          "resource" : "morphologyService"
        }
      },
      "template"  : "templates/morph3.html",
      "contextMenu" : true,
      "contextMenuTemplate": "templates/arethusa.morph/context_menu.html",
      "lexicalInventory" : {
        "retriever" : {
          "LexicalInventoryRetriever" : {
            "resource" : "lexInvFusekiEndpoint"
          }
        }
      },
      "gloss" : true,
      "matchAll" : true,
      "@include" : "morph/gr_attributes2.json"
    },
```
If you set the auxConfPath in the staging.json as follows, this should work, because the live smyth3.json there does have the gloss attribute turned on

```
 "auxConfPath": "http://services.perseids.org/arethusa-configs",
```

If instead you built the config files locally, or used them from github, this might not work, because I think only the version of smyth3.json in the gcelano branch of arethusa-configs has gloss turned on right now, and what is in the GitHub dist directory might be built from master.  (We really need to merge all the branches in arethusa-configs back into master but that is another story...)

As for the access deined message when using perseids config, it's because it's trying to access a remote session on the perseids server, and you cannot do that locally due to the cors restrictions.

**Comment by Masoumeh** [comment]
I think the problem was caused by "not" passing the variable "form"
as the second argument of the function "updateGloss". I create a pull request.

**Comment by balmas** [comment]
Fixed in ae7f5b0
